lib/stripe/invoice.rb

```diff
@@ -4,28 +4,45 @@ module Stripe
   class Invoice < APIResource
     extend Stripe::APIOperations::List
     include Stripe::APIOperations::Save
+    include Stripe::APIOperations::Delete
     extend Stripe::APIOperations::Create
 
     OBJECT_NAME = "invoice".freeze
 
-    def self.upcoming(params, opts = {})
-      resp, opts = request(:get, upcoming_url, params, opts)
-      Util.convert_to_stripe_object(resp.data, opts)
+    def finalize(params = {}, opts = {})
+      url = resource_url + "/finalize"
+      resp, opts = request(:post, url, params, opts)
+      initialize_from(resp.data, opts)
+    end
+
+    def mark_uncollectible(params = {}, opts = {})
+      url = resource_url + "/mark_uncollectible"
+      resp, opts = request(:post, url, params, opts)
+      initialize_from(resp.data, opts)
     end
 
     def pay(params = {}, opts = {})
-      resp, opts = request(:post, pay_url, params, opts)
+      url = resource_url + "/pay"
+      resp, opts = request(:post, url, params, opts)
       initialize_from(resp.data, opts)
     end
 
-    def self.upcoming_url
-      resource_url + "/upcoming"
+    def send(params = {}, opts = {})
+      url = resource_url + "/send"
+      resp, opts = request(:post, url, params, opts)
+      initialize_from(resp.data, opts)
+    end
```

Comment:
We need a way to override `send` as it's a default ruby method. Or come up with a new name instead of the hack.